### ISSUE TITLE
Can't see worker node

### ISSUE THREAD

**Issue by dza89** [k0sproject/k0s#2170]
### Before creating an issue, make sure you've checked the following:

- [X] You are running the latest released version of k0s
- [X] Make sure you've searched for existing issues, both open and closed
- [X] Make sure you've searched for PRs too, a fix might've been merged already
- [X] You're looking at docs for the released version, "main" branch docs are usually ahead of released versions.

### Platform

```shell
Linux 5.15.68-1-lts #1 SMP Thu, 15 Sep 2022 09:53:50 +0000 x86_64 GNU/Linux
NAME="Arch Linux"
PRETTY_NAME="Arch Linux"
ID=arch
BUILD_ID=rolling
ANSI_COLOR="38;2;23;147;209"
HOME_URL="https://archlinux.org/"
DOCUMENTATION_URL="https://wiki.archlinux.org/"
SUPPORT_URL="https://bbs.archlinux.org/"
BUG_REPORT_URL="https://bugs.archlinux.org/"
LOGO=archlinux-logo

```

### Version

v1.24.4+k0s.0

### Sysinfo
```shell
Machine ID: "042798b16a2e78c8ee20a4e119d44b23dcfa783916d4868f5134b049dcc1efd4" (from machine) (pass)
Total memory: 15.4 GiB (pass)
Disk space available for /var/lib/k0s: 217.9 GiB (pass)
Operating system: Linux (pass)
  Linux kernel release: 5.15.68-1-lts (pass)
  Max. file descriptors per process: current: 1024 / max: 524288 (warning: < 65536)
  Executable in path: modprobe: /usr/bin/modprobe (pass)
  /proc file system: mounted (0x9fa0) (pass)
  Control Groups: version 2 (pass)
    cgroup controller "cpu": available (pass)
    cgroup controller "cpuacct": available (via cpu in version 2) (pass)
    cgroup controller "cpuset": available (pass)
    cgroup controller "memory": available (pass)
    cgroup controller "devices": available (assumed) (pass)
    cgroup controller "freezer": available (assumed) (pass)
    cgroup controller "pids": available (pass)
    cgroup controller "hugetlb": available (pass)
    cgroup controller "blkio": available (via io in version 2) (pass)
  CONFIG_CGROUPS: Control Group support: built-in (pass)
    CONFIG_CGROUP_FREEZER: Freezer cgroup subsystem: built-in (pass)
    CONFIG_CGROUP_PIDS: PIDs cgroup subsystem: built-in (pass)
    CONFIG_CGROUP_DEVICE: Device controller for cgroups: built-in (pass)
    CONFIG_CPUSETS: Cpuset support: built-in (pass)
    CONFIG_CGROUP_CPUACCT: Simple CPU accounting cgroup subsystem: built-in (pass)
    CONFIG_MEMCG: Memory Resource Controller for Control Groups: built-in (pass)
    CONFIG_CGROUP_HUGETLB: HugeTLB Resource Controller for Control Groups: built-in (pass)
    CONFIG_CGROUP_SCHED: Group CPU scheduler: built-in (pass)
      CONFIG_FAIR_GROUP_SCHED: Group scheduling for SCHED_OTHER: built-in (pass)
        CONFIG_CFS_BANDWIDTH: CPU bandwidth provisioning for FAIR_GROUP_SCHED: built-in (pass)
    CONFIG_BLK_CGROUP: Block IO controller: built-in (pass)
  CONFIG_NAMESPACES: Namespaces support: built-in (pass)
    CONFIG_UTS_NS: UTS namespace: built-in (pass)
    CONFIG_IPC_NS: IPC namespace: built-in (pass)
    CONFIG_PID_NS: PID namespace: built-in (pass)
    CONFIG_NET_NS: Network namespace: built-in (pass)
  CONFIG_NET: Networking support: built-in (pass)
    CONFIG_INET: TCP/IP networking: built-in (pass)
      CONFIG_IPV6: The IPv6 protocol: built-in (pass)
    CONFIG_NETFILTER: Network packet filtering framework (Netfilter): built-in (pass)
      CONFIG_NETFILTER_ADVANCED: Advanced netfilter configuration: built-in (pass)
      CONFIG_NETFILTER_XTABLES: Netfilter Xtables support: module (pass)
        CONFIG_NETFILTER_XT_TARGET_REDIRECT: REDIRECT target support: module (pass)
        CONFIG_NETFILTER_XT_MATCH_COMMENT: "comment" match support: module (pass)
        CONFIG_NETFILTER_XT_MARK: nfmark target and match support: module (pass)
        CONFIG_NETFILTER_XT_SET: set target and match support: module (pass)
        CONFIG_NETFILTER_XT_TARGET_MASQUERADE: MASQUERADE target support: module (pass)
        CONFIG_NETFILTER_XT_NAT: "SNAT and DNAT" targets support: module (pass)
        CONFIG_NETFILTER_XT_MATCH_ADDRTYPE: "addrtype" address type match support: module (pass)
        CONFIG_NETFILTER_XT_MATCH_CONNTRACK: "conntrack" connection tracking match support: module (pass)
        CONFIG_NETFILTER_XT_MATCH_MULTIPORT: "multiport" Multiple port match support: module (pass)
        CONFIG_NETFILTER_XT_MATCH_RECENT: "recent" match support: module (pass)
        CONFIG_NETFILTER_XT_MATCH_STATISTIC: "statistic" match support: module (pass)
      CONFIG_NETFILTER_NETLINK: module (pass)
      CONFIG_NF_CONNTRACK: Netfilter connection tracking support: module (pass)
      CONFIG_NF_NAT: module (pass)
      CONFIG_IP_SET: IP set support: module (pass)
        CONFIG_IP_SET_HASH_IP: hash:ip set support: module (pass)
        CONFIG_IP_SET_HASH_NET: hash:net set support: module (pass)
      CONFIG_IP_VS: IP virtual server support: module (pass)
        CONFIG_IP_VS_NFCT: Netfilter connection tracking: built-in (pass)
      CONFIG_NF_CONNTRACK_IPV4: IPv4 connetion tracking support (required for NAT): unknown (warning)
      CONFIG_NF_REJECT_IPV4: IPv4 packet rejection: module (pass)
      CONFIG_NF_NAT_IPV4: IPv4 NAT: unknown (warning)
      CONFIG_IP_NF_IPTABLES: IP tables support: module (pass)
        CONFIG_IP_NF_FILTER: Packet filtering: module (pass)
          CONFIG_IP_NF_TARGET_REJECT: REJECT target support: module (pass)
        CONFIG_IP_NF_NAT: iptables NAT support: module (pass)
        CONFIG_IP_NF_MANGLE: Packet mangling: module (pass)
      CONFIG_NF_DEFRAG_IPV4: module (pass)
      CONFIG_NF_CONNTRACK_IPV6: IPv6 connetion tracking support (required for NAT): unknown (warning)
      CONFIG_NF_NAT_IPV6: IPv6 NAT: unknown (warning)
      CONFIG_IP6_NF_IPTABLES: IP6 tables support: module (pass)
        CONFIG_IP6_NF_FILTER: Packet filtering: module (pass)
        CONFIG_IP6_NF_MANGLE: Packet mangling: module (pass)
        CONFIG_IP6_NF_NAT: ip6tables NAT support: module (pass)
      CONFIG_NF_DEFRAG_IPV6: module (pass)
    CONFIG_BRIDGE: 802.1d Ethernet Bridging: module (pass)
      CONFIG_LLC: module (pass)
      CONFIG_STP: module (pass)
  CONFIG_EXT4_FS: The Extended 4 (ext4) filesystem: module (pass)
  CONFIG_PROC_FS: /proc file system support: built-in (pass)
```

### What happened?

I've accidentally applied my k0sctl config with role: controller where I should have applied controller+worker.
I've reset and rebooted the node and applied the controller+worker, but the worker still doesn't show up.

### Steps to reproduce

1. apply a HA oonfig with a role: controller
2. k0s stop && k0s reset && reboot (i assume this is the way for graceful removal but not sure)
3. apply same HA config but swap role to controller+worker
4. worker will not show

### Expected behavior
`
kubectl get nodes
`
shows me the new worker.

### Actual behavior

`
kubectl get nodes
`
shows me no new worker

### Screenshots and logs

k0s status:
```
Version: v1.24.4+k0s.0
Process ID: 916
Role: controller
Workloads: true
SingleNode: false
```

Looks good to me...

### Additional context

Looks a bit like https://github.com/k0sproject/k0s/issues/1996

**Comment by dza89** [comment]
Ok, after k0s stop and reset see the node still in 
```
k0s etcd member-list
```

Do i need to manually remove it from there?

**Comment by jnummelin** [comment]
```
Role: controller
```

This indicates that it's still running as plain controller. How are you running k0s, via systemd? If so what does the systemd unit look like? Or if not systemd, how are you launching k0s?

To have the controller to run workloads too, you need either `--enable-worker` or `--single`. The latter implies a special single node case, i.e. no other nodes can be joined to the cluster effectively.

**Comment by dza89** [comment]
The problem is that even when i do a k0s reset, the node isn't removed from etcd.
So whenever 

> ```
> Role: controller
> ```
> 
> This indicates that it's still running as plain controller. How are you running k0s, via systemd? If so what does the systemd unit look like? Or if not systemd, how are you launching k0s?

I use k0sctl.

> 
> To have the controller to run workloads too, you need either `--enable-worker` or `--single`. The latter implies a special single node case, i.e. no other nodes can be joined to the cluster effectively.

The problem is whenever a node has had the controller role, even k0sctl says: can't switch roles from controller to controller+worker. 
So my assumption was: alright, I'll just purge the entire node from the cluster, but even that doesn't seem to be possible at all, as the node stays registered in etcd. 
So basically: how do i successfully remove a node from a cluster?

**Comment by jnummelin** [comment]
removing a controller+worker node from kube api does not indeed remove it from etcd cluster. that has to be done manually. so the process is like: (assuming you've got nodes A,B and C and want to remove C)
0. cordon & drain node C via kube api
1. On node C: k0s stop && k0s reset
2. On node A or B: k0s kc delete node C
3. On node A or B: k0s etcd leave --peer-address 1.2.3.4 (the peer address for C)

We used to have docs around this, but now looking through the docs site it seems they got lost in some refactoring 🤦

**Comment by dza89** [comment]
I see https://github.com/k0sproject/k0s/issues/2181 now.
Thanks.
